### PR TITLE
Add Storage class

### LINKED ISSUE #69: Add Storage class

### PR DESCRIPTION
**Pull request by jinzihan2002**
Fixes #69 

Adds Storage class and subclasses to load student, component and grade lists from data files, and save to them.
Current implementation will work on the happy path, with no data corruption.

Parameterised constructors have been added to studentList, componentList and gradeList classes to initialise their ArrayLists.

Storage will be made more robust to handle data corruption in a future issue.

### CODE REVIEW

**Review by ThienDuc3112 — APPROVED** [review]
Each individual throwing IOException instead of our own exception is a bit sketchy, but other than that LGTM

File: src/main/java/tutorlink/storage/ComponentStorage.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            return new Exam(name, maxScore, weight);
+        default:
+            throw new RuntimeException();
+        }
+    }
+
+    private String getFileInputForComponent(Component component) {
+        String convertedString = "";
+        if (component instanceof ClassParticipation) {
+            convertedString += "P | ";
```
Comment (RCPilot1604) [inline src/main/java/tutorlink/storage/ComponentStorage.java]: For the sake of code quality, consider extracting these magic literals?

File: src/main/java/tutorlink/storage/GradeStorage.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                if (student.getMatricNumber().equals(matricNumber)) {
+                    selectedStudent = student;
+                    break;
+                }
+            }
+
+            if (selectedComp != null && selectedStudent != null) {
+                Grade newGrade = new Grade(selectedComp, selectedStudent, score);
+                grades.add(newGrade);
+            }
```
Comment (RCPilot1604) [inline src/main/java/tutorlink/storage/GradeStorage.java:50]: I like how you check that for each grade in the gradelist, the referenced student and component exists in their respective component lists. 

We will have to explain this in our UG.

**Review by RCPilot1604 — APPROVED** [review]
Looks good to me.

**Pull request merged**